### PR TITLE
[MM 13161] Mobile: Add paging to Select Teams screen

### PR DESCRIPTION
**Pull request by migbot**
#### Summary
The FlatList component in SelectTeam has been updated to support infinite scrolling of paginated teams. Because of pagination, sorting of the teams on the front end side has been removed so that the user does not experience list item rearrangement as more teams get loaded. Refresh support has been added to reset the page to 0 so that open teams that may have been added after all teams had been loaded are fetched. The new results are displayed at the end of the list so the user must still scroll down after the refresh. We might consider sorting on the backend by most recently created.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/9951

#### Checklist
- [x] Added or updated unit tests (required for all new features)
- [x] All new/modified APIs include changes to [mattermost-redux](https://github.com/mattermost/mattermost-redux): https://github.com/mattermost/mattermost-redux/pull/742
- [x] Has UI changes

#### Device Information
This PR was tested on:
* Android emulator
* LG K20V running Android 7.0
* iOS emulator
* iPhone 8 running iOS 12.1

#### Screenshots
Refresh
refresh_android

Load More
load_more_android

### CODE REVIEW

**Comment by hmhealey** [comment]
Our API doesn't support pagination well at this time. The way we do it in other places is that if we receive less than a full page of results, we know that it's the last page. It can lead to some weird cases when there's exactly 1 page of results (or a multiple thereof), but that usually works. You'd just need to keep track of whether or not we've hit that point to prevent calling `getTeams` again.

Also, I assume you're just passing `extraData={this.state.loading}`, but if you're having to pass multiple values into it, we have a helper function that can help with that, although it should be avoided if possible

**Comment by enahum** [comment]
@migbot you can resolve that easily by using `onScroll` instead of `onEndReached` I'm definitely not a fan of `onEndReached` at all

**Comment by esethna** [comment]
Thanks @migbot! @hmhealey @enahum is this ready to trigger a build for testing?

**Comment by hanzei** [comment]
Hi @migbot,

This PR needs a rebase. After this we can review and hopefully merge it. If you need help with the rebase, let me know.

**Comment by hanzei** [comment]
@enahum @esethna Could you give this a review?

**Comment by enahum** [comment]
There is an issue with the build server that for some reason mattermost-redux is not getting prepared, before building we need to ssh to the box, I’ll do that today

**Comment by enahum** [comment]
@migbot could you please rebase this PR once more?

**Comment by esethna** [comment]
@enahum when the build is fixed can you please trigger a test build for this PR?

**Comment by hanzei** [comment]
@enahum Could you please take care of triggering a test build for this PR?

**Comment by enahum** [comment]
CN we give this another rebase first?

**Comment by hmhealey** [comment]
Does the build bot not merge before building? The branch has no conflicts, so I would assume it should merge and build fine

**Comment by enahum** [comment]
@hmhealey I guess they wanted to test the feature before merging, but surely we can merge it into master and test it with the next scheduled build

**Comment by hmhealey** [comment]
Let me rephrase that since I think I was unclear. For the web app and server, Mattermod merges the PR and master branches, and the then, the resulting code is deployed to the test server so that you can test the code as if the PR had actually been merged. Does it not do that for mobile apps as well?

**Comment by enahum** [comment]
@migbot I think @esethna wanted to test it, so I'm going to build the PR so they can test it properly. I'll start the build in a about 30 mins

**Comment by enahum** [comment]
building manually

**Comment by enahum** [comment]
@migbot once @esethna finishes with the testing of this PR you can rebase it and if there is nothing else to do with it then we can merge it

**Comment by esethna** [comment]
@migbot please help rebase and we can merge

**Comment by esethna** [comment]
No labels required, thanks Miguel
